tests/unit/sysinfo_test.c

```diff
@@ -13,12 +14,56 @@ static void test_uptime(void)
     assert_in_range(uptime, 1, 60*24*365*5);
 }
 
+static void test_extract_version_number(void)
+{
+    char version_str[CF_BUFSIZE];
+    char *major;
+    char *minor;
+    char *patch;
+    char *bogus;
+    char *next;
+    
+    strcpy(version_str, "Ubuntu 20.04.1 LTS");
```

Comment:
Missing some edge cases here:

```
""
" "
" no numbers in sight "
"0"
"1k"
"1234"
"1 2 3 4"
"Debian 9"
"Cent OS 6.7"
"CFEngine 3.18.0-2deadbeef"
```

Tips: You could make a wrapper for the test which calls the function in a loop, adding each number to a `Seq`. Then it's easier and less repetitive to test more cases.